src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -791,7 +807,7 @@ internal async Task HandleCopyResultsRequest(CopyResultsRequestParams requestPar
                             }
                         }
                         // Add line break if this is not the last row in all selections.
-                        if (rowIndex + pageStartRowIndex != lastRowIndex && (!builder.ToString().EndsWith(Environment.NewLine) || !Settings.QueryEditorSettings.Results.SkipNewLineAfterTrailingLineBreak))
+                        if (rowIndex + pageStartRowIndex != lastRowIndex && (!builder.ToString().EndsWith(Environment.NewLine) || (!Settings?.QueryEditorSettings?.Results?.SkipNewLineAfterTrailingLineBreak ?? true)))
```

Comment:
In the issues we've seen, `QueryEditorSettings` itself are null, as they would fail to initialize. ref. https://github.com/microsoft/azuredatastudio/issues/24476#issuecomment-1732225384 I will add a check to log it properly elsewhere, here we're only handling possible nulls. I'm working on another PR to make more improvements on how settings are loading/updating, so will take this feedback in there.